packages/ai-api/src/utils/deployment-resolver.ts

```diff
@@ -46,8 +46,17 @@ export interface ResourceGroupConfiguration {
  */
 export type ModelDeployment<ModelNameT = string> =
   | ModelNameT
-  | ((ModelConfiguration<ModelNameT> | DeploymentIdConfiguration) &
-      ResourceGroupConfiguration);
+  | ConfigurationOptions<ModelNameT>;
+
+/**
+ * The configuration options for a model deployment.
+ * @typeParam ModelNameT - String literal type representing the name of the model.
+ */
+export type ConfigurationOptions<ModelNameT> = Xor<
```

Comment:
[req] This name is not speaking at all, could also be `Option` or `Configuration` or `OptionsConfiguration` - aka. quite random ;)
Consider something like `ModelDeploymentConfig` or `ModelDeploymentDetails` or similar.

Reply:
Yes agreed, I was also not satisfied with the name, but there were so many renamings that I wanted to wait until somethings sticks before I change it (again) 😬